code_mirror/toit.js

```diff
@@ -45,12 +45,12 @@
     var atoms = makeJsObject("true|false|null");
     var specialVars = makeJsObject("this|super|it");
 
-    var IDENTIFIER = /[a-zA-Z_]\w*/;
-    var TYPE = /[a-zA-Z_]\w*(\.[a-zA-Z_]\w*)?/;
+    var IDENTIFIER = /[a-zA-Z_][\w-]*/;
+    var TYPE = /[a-zA-Z_][\w-]*(\.[a-zA-Z_][\w-]*)?/;
     var OVERRIDABLE_OPERATOR = /==|>=|<=|<<|>>>|>>|\*|\+|-|%|\/|<|>|&|\||\^|~|\[\]\=|\[\]|\[\.\.\]/
 
     var CONSTANT_HEURISTIC = /_?[A-Z][A-Z_0-9]+/;
```

Comment:
We can have dashes in constants too, no?

Reply:
good catch.
I also fixed the heuristics: they weren't taking the full input into account.